### PR TITLE
Arranging Ingredients Input Screens

### PR DESCRIPTION
**Pull request by mche3kek**
Adding an intermediate screen used to display the picture taken / chosen from the gallery. This screen contains the Barcode and TextRecognition buttons and returns to the camera screen once the analyzing is done displaying either an error message or a success message. This modification is for a more user-friendly and understandable experience instead of the previous one where the user doesn't understand that he needs to take a picture before pressing on the analyzing button.

### CODE REVIEW

File: app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt [hunk 1]
```diff
@@ -324,4 +315,18 @@ class CameraViewModel : ViewModel() {
...
+   */
+  fun empty() {
+    _errorToDisplay.value = null
+    _informationToDisplay.value = null
+  }
+
+  /** This function is used to empty the list of ingredients after it has been processed. */
+  fun emptyIngredients() {
+    _listOfIngredientToInput.value = emptyList()
+  }
```
Comment (ohamz) [inline app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt:331]: Nice idea to add inner boolean values, to be sure nothing is executed before the screen is displayed.

File: app/src/main/java/com/android/feedme/ui/camera/Camera.kt [hunk 2]
```diff
@@ -82,38 +67,27 @@ import com.android.feedme.ui.theme.CameraButtonsBackground
  * images and viewing them in a gallery. Utilizes CameraX for camera operations and Jetpack Compose
  * for the UI components.
  */
-@OptIn(ExperimentalMaterial3Api::class)
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalCoroutinesApi::class)
```
Comment (ohamz) [inline app/src/main/java/com/android/feedme/ui/camera/Camera.kt:70]: What does this add-on do ?

File: app/src/main/java/com/android/feedme/ui/camera/Camera.kt [hunk 3]
```diff
@@ -321,28 +280,3 @@ fun CameraPreview(controller: LifecycleCameraController, modifier: Modifier = Mo
...
-        modifier = modifier) {
-          items(bitmaps) { bitmap ->
-            Image(
-                bitmap = bitmap.asImageBitmap(),
-                contentDescription = "Photo",
-                modifier = Modifier.clip(RoundedCornerShape(10.dp)))
-          }
-        }
-  }
-}
```
Comment (ohamz) [inline app/src/main/java/com/android/feedme/ui/camera/Camera.kt:348]: So the local gallery feature was totally removed right ?

**Review by ohamz — CHANGES_REQUESTED** [review]
Great coverage ! Implementing and testing this issue was not simple, well done!

I like the new division of the buttons through 2 screens, and the navigation is smooth.

I only have a few questions below.

File: app/src/main/java/com/android/feedme/ui/camera/Camera.kt (hunk 3 above)
Comment (mche3kek) [reply app/src/main/java/com/android/feedme/ui/camera/Camera.kt:348]: Yes indeed, it was not needed now that we process pictures one by one

File: app/src/main/java/com/android/feedme/ui/camera/Camera.kt (hunk 2 above)
Comment (mche3kek) [reply app/src/main/java/com/android/feedme/ui/camera/Camera.kt:70]: It was a remnant of something I tried using coroutines to wait for the result of the photo before navigating to the second screen, before settling with using Booleans. I will remove it :)

**Review by ohamz — APPROVED** [review]
Great. Approved !

**Pull request merged**